### ISSUE TITLE
[filer] delete folder via filer is inconsistent with weed mount

### ISSUE THREAD

**Issue by paochiang** [chrislusf/seaweedfs#2108]
**Describe the bug**
After deleting the folder via filer api, although the folder is not seen in the `weed mount`, but we cannot create the same folder immediately.

**System Setup**
- List the command line to start "weed master", "weed volume", "weed filer", "weed s3", "weed mount".
```
# setup master
weed master -port=9333 -mdir=/tmp/master0 -defaultReplication=000 -ip=localhost
# setup volume
weed volume -port=8080 -dir=/tmp/volume0 -ip=localhost -mserver=localhost:9333
# setup filer
weed filer -port=8888  -ip=localhost -master=localhost:9333
#weed mount
weed mount -filer=localhost:8888 -cacheCapacityMB=0 -cacheDir=/tmp/cache001 -dir=/root/mnt

# mkdir and ls
root@dev-4:~# mkdir -p /root/mnt/folder1/sub1
root@dev-4:~# ls /root/mnt/folder1
sub1

# delete folder via filer and  ls，and  create the same folder immediately  gets error
root@dev-4:~# curl -X DELETE http://localhost:8888/folder1/sub1?recursive=true
root@dev-4:~# ls /root/mnt/folder1
root@dev-4:~# mkdir -p /root/mnt/folder1/sub1
mkdir: cannot create directory ‘/root/mnt/folder1/sub1’: File exists

btw, if filer store use mongodb, I cannot delete an empty folder via filer(curl -X DELETE http://localhost:8888/folder1/sub1?recursive=true),  
it is not as expected.

# Waiting for a while to create the same folder is ok 
root@dev-4:~# mkdir -p /root/mnt/folder1/sub1
root@dev-4:~# ls /root/mnt/folder1
sub1
```
- OS version
`Ubuntu 18.04.4 LTS`
- output of `weed version`
`version 30GB 2.48 45a7622 linux amd64`
- if using filer, show the content of `filer.toml`
default

**Expected behavior**
1、After deleting the folder through the filer, we can create the same folder immediately.
2、We can delete empty folder through the filer if filer store is mongodb.

**Screenshots**
![image](https://user-images.githubusercontent.com/28077875/120805007-d9c0bf80-c577-11eb-842b-0d931ac0320d.png)
**Additional context**

**Comment by chrislusf** [comment]
> # Waiting for a while to create the same folder is ok

How long? ms/second/minute/hour?

**Comment by chrislusf** [comment]
for mongodb, can not reproduce.

**Comment by paochiang** [comment]
> > Waiting for a while to create the same folder is ok
> 
> How long? ms/second/minute/hour?

minute level, in my test, it takes about 1 minute.
I exec the below shell after `weed mount`:
test.sh:
```
#!/bin/bash

mkdir -p /root/mnt/folder1/sub1
curl -X DELETE http://localhost:8888/folder1/sub1?recursive=true
for((i=0; i!=1;))
do
    mkdir /root/mnt/folder1/sub1
    if [ $? -ne 0 ]; then
        echo "failed"
    else
        i=1
    fi
done
echo "succeed"
```
exec shell:
```
time test.sh
```
output:
```
real	1m0.017s
user	0m31.267s
sys	0m30.741s
```

**Comment by chrislusf** [comment]
do you have a docker compose file to reproduce this?

the mount should receive metadata updates fairly quickly. You can use `weed -vmodule=meta_cache_subscribe=4 mount` to see when the metadata updates are received. for example:

```
tmp$ weed -vmodule=meta_cache_subscribe=4 mount -dir=mm
This is SeaweedFS version 30GB 2.50  darwin amd64
mount point owner uid=501 gid=20 mode=drwxr-xr-x
I0604 21:10:05 51080 leveldb_store.go:41] filer store dir: /var/folders/86/3zk5ylp176d6vtrjxqst7f280000gn/T/761d035e/meta
I0604 21:10:05 51080 file_util.go:23] Folder /var/folders/86/3zk5ylp176d6vtrjxqst7f280000gn/T/761d035e/meta Permission: -rwxr-xr-x
I0604 21:10:06 51080 mount_std.go:246] mounted localhost:8888/ to mm
I0604 21:10:58 51080 meta_cache_subscribe.go:31] deleting /docker/nginx/proxy.conf
I0604 21:10:58 51080 meta_cache_subscribe.go:31] deleting /docker/nginx

```

**Comment by paochiang** [comment]
> do you have a docker compose file to reproduce this?

local-mount-compose.yml:
```
version: '2'

services:
  master:
    image: chrislusf/seaweedfs:2.48
    ports:
      - 9333:9333
      - 19333:19333
    command: "master -ip=master"
  volume:
    image: chrislusf/seaweedfs:2.48
    ports:
      - 7455:8080
      - 9325:9325
    command: 'volume -mserver="master:9333" -port=8080 -metricsPort=9325 -preStopSeconds=1 -publicUrl=localhost:7455'
    depends_on:
      - master
  filer:
    image: chrislusf/seaweedfs:2.48
    ports:
      - 8888:8888
      - 18888:18888
      - 9326:9326
    command: 'filer -master="master:9333"  -metricsPort=9326'
    tty: true
    stdin_open: true
    depends_on:
      - master
      - volume
  mount_1:
    image: chrislusf/seaweedfs:2.48
    privileged: true
    entrypoint: '/bin/sh -c "mkdir -p t1 && mkdir -p cache/t1 && weed -v=4 -vmodule=meta_cache_subscribe=4 mount  -filer=filer:8888 -cacheDir=./cache/t1 -dir=./t1 -filer.path=/c1"'
    depends_on:
      - master
      - volume
      - filer
```

**setup docker-compse:**
```
docker-compose -f ./local-mount-compose.yml up
```

**enter the mount container**:
```
docker exec -it desktop_mount_1_1 sh
```
**install curl**:
```
apk add curl
```

**create a shell script**:(test.sh)
```
#!/bin/sh
mkdir ./t1/folder1
echo `curl -X DELETE http://filer:8888/c1/folder1?recursive=true`

while true
do
    mkdir ./t1/folder1
    if [ $? -ne 0 ]; then
        echo "failed"
        echo "ls result:"`ls ./t1/`
    else
        echo "succeed"
        exit 0
    fi
done
```

**exec shell**:
```
time ./test.sh
```

**output**:
```
................   omit
mkdir: can't create directory './t1/folder1': File exists
failed
ls result:
mkdir: can't create directory './t1/folder1': File exists
failed
ls result:
mkdir: can't create directory './t1/folder1': File exists
failed
ls result:
mkdir: can't create directory './t1/folder1': File exists
failed
ls result:
succeed
real	1m 0.00s
user	0m 18.93s
sys	0m 19.69s
```

**Comment by paochiang** [comment]
> for mongodb, can not reproduce.

It's strange, I also can't reproduce it in new env, but in one my running env, this problem has always existed.

**Comment by paochiang** [comment]
> the mount should receive metadata updates fairly quickly.

After exec `curl -X DELETE http://filer:8888/path/to/dir?recursive=true`, in `weed mount`, `ls` command does not see the deleted folder(I think that `weed mount` has received metadata at this time), but cannot create the same folder.

**Comment by chrislusf** [comment]
Thanks for the detailed steps! Added the fix.

**Comment by chrislusf** [comment]
> > for mongodb, can not reproduce.
> 
> It's strange, I also can't reproduce it in new env, but in one my running env, this problem has always existed.

Added a possible fix in 21ad9a4ac29fad0794369cc05d1f29fa91340c34

**Comment by paochiang** [comment]
> > > for mongodb, can not reproduce.
> > 
> > 
> > It's strange, I also can't reproduce it in new env, but in one my running env, this problem has always existed.
> 
> Added a possible fix in [21ad9a4](https://github.com/chrislusf/seaweedfs/commit/21ad9a4ac29fad0794369cc05d1f29fa91340c34)

I did some inspections again.（cannot delete empty folder: `/pavostor/gemini/traindata/skcw5wn4m8an/829`）

After deleting empty folder via filer api.

**`weed mount` check:**
![image](https://user-images.githubusercontent.com/28077875/120953778-56be8580-c780-11eb-8ce4-6d57ca15973a.png)
we can see:
1、folder `829` is still existed.
2、there are total 1418 folders under  folder `/pavostor/gemini/traindata/skcw5wn4m8an`

**mongodb check:** 
![image](https://user-images.githubusercontent.com/28077875/120953704-28d94100-c780-11eb-86d2-15e09a0c4890.png)
we can see:
1、folder `829` is not existed. (data in filer store is correct)
2、there are total 1406 folders under  folder `/pavostor/gemini/traindata/skcw5wn4m8an` (So there are more than a dozen of these unremovable folders)

So, I think it may be because the network is relatively poor, causing the metadata of the filer delete folder without  successful 
 synchronizing to the `weed mount`, causing the data of the `weed mount` and the filer store inconsistent.

**Comment by chrislusf** [comment]
not really. It was because metadata updates were not successfully applied to fuse internal objects.
